src/test/java/test/retryAnalyzer/RetryAnalyzerTest.java

```diff
@@ -41,4 +43,16 @@ public void testIfRetryIsInvokedBeforeListener() {
         tng.run();
         assertThat(TestClassSample.messages).containsExactly("afterInvocation", "retry", "afterInvocation");
     }
+
+    @Test(description = "GITHUB-1600")
+    public void testIfRetryIsInvokedBeforeListenerButHasToConsiderFailures() {
+        TestNG tng = create(Github1600TestSample.class);
+        Github1600Listener listener = new Github1600Listener();
+        TestListenerAdapter tla = new TestListenerAdapter();
+        tng.addListener((ITestNGListener) tla);
+        tng.addListener((ITestNGListener) listener);
```

Comment:
I'm just thinking a `TestNG#addListeners(ITestNGListener... listeners)` could be awesome.

Reply:
Its use would be limited to just us for writing our unit tests, and to those that use the TestNG APIs directly (perhaps the IDE plugins and the build tool plugins). I dont see much of a use for this in other places. But yes, I agree, it would be simple change.